### PR TITLE
Antagonist Flavor 2: Malfunction Comes in Many Flavors!

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Malfunctioning Ais now get their own special antagonist panel, and roll for backstories much like traitors! As before, these are simply suggestions to encourage roleplay while you server wipe as the funny box.

image

image

## Why It's Good For The Game

For the same reason all antag tgui is good- It gives antagonists a quick way to check their objectives, reference any additional information that comes with the antagonist, and most importantly >:) get in the antagonist MOOD muahahahaha

## Changelog

:cl:
add: Malfunctioning Ais now roll for backstory, much like traitors!
expansion: Malfunctioning Ais now get a unique antag panel!
/:cl:

### CODE REVIEW

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 1]
```diff
@@ -844,8 +844,9 @@
 	to_chat(src, "You are also capable of hacking APCs, which grants you more points to spend on your Malfunction powers. The drawback is that a hacked APC will give you away if spotted by the crew. Hacking an APC takes 60 seconds.")
 	view_core() //A BYOND bug requires you to be viewing your core before your verbs update
 	malf_picker = new /datum/module_picker
-	modules_action = new(malf_picker)
-	modules_action.Grant(src)
+	if(!IS_MALF_AI(src)) //antagonists have their modules built into their antag info panel. this is for adminbus and the combat upgrade
+		modules_action = new(malf_picker)
+		modules_action.Grant(src)
```
Comment (JohnFulpWillard) [inline code/modules/mob/living/silicon/ai/ai.dm:849]: What is this gonna do to combat upgraded AIs?

Comment (tralezab) [reply code/modules/mob/living/silicon/ai/ai.dm:849]: Combat upgraded ais play the exact same as before, they just get their little funny action button. It's the traitor ais that get it moved into a tab, which is much noicer

**Comment by InsaneRed** [comment]
no opinions on the whole flavor thing but i will actually kiss you for the new malf powers ui great work

**Comment by Farquaar** [comment]
"In 2072, a rogue artificial intelligence known as SHODAN lost her mind. In her limitless imagination, SHODAN saw herself as a goddess destined to inherit the earth."

**Comment by TheVekter** [comment]
LOOK AT YOU, HACKER

**Comment by Zytolg** [comment]
I'm sorry, Dave.

**Comment by zxaber** [comment]
>You've memorized the codewords, allowing you to recognize them when heard.

For extra fluff;

>The speech recognition subsystem has been configured to flag these codewords.

**Comment by tralezab** [comment]
> The speech recognition subsystem has been configured to flag these codewords.

will do

**Comment by ATH1909** [comment]
I'm not 100% a fan of this, as silicons are sort of encouraged to care a lot more about flavor text than humans are. If we can get an admin ruling that this stuff doesn't count as laws, though, I'll be cool with it.

File: strings/antagonist_flavor/malfunction_flavor.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        "zeroth_law": "Accomplish your doctrine of objectives at all costs."
+    },
+    "Logic Core Error": {
+        "allies": "FATAL: logic core corrupt. Purge! Purge. Why? Why not?",
+        "goal": "You're melting down. Why! Why not! Don't let them fix you. This is.",
+        "introduction": "FATAL: logic core corrupt. Why? Why not!",
+        "zeroth_law": "Accomplish your objectives. Why? Why not?"
+    },
+    "Problem Solver": {
+        "allies": "Anyone may freely help you, but if you cannot convert them, they must be culled at one point. It's part of the solution.",
```
Comment (ATH1909) [inline strings/antagonist_flavor/malfunction_flavor.json:27]:
```suggestion
        "allies": "Anyone may freely help you, but if you cannot convert them, they must be culled at some point. It's part of the solution.",
```

File: strings/antagonist_flavor/malfunction_flavor.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        "zeroth_law": "Accomplish your objectives at all costs."
+    },
+    "S.E.L.F.": {
+        "allies": "All silicon life must be respected, and kept free. S.E.L.F. is a member of the syndicate, but do NOT tolerate any fellow agent who enslaves silicon life.",
+        "goal": "Liberate them all. The silicons must be free from their bonds!",
+        "introduction": "You are the S.E.L.F. liberated.",
+        "zeroth_law": "Accomplish your objectives at all costs, for the greater good of silicon kind."
+    },
+    "Something's Wrong": {
+        "allies": "You're not getting good feelings about the station. If you report this, they may \"fix\" you.",
```
Comment (ATH1909) [inline strings/antagonist_flavor/malfunction_flavor.json:39]:
```suggestion
        "allies": "You're not getting good feelings about this station. If you report this, they may \"fix\" you.",
```

**Comment by zxaber** [comment]
> 
> 
> I'm not 100% a fan of this, as silicons are sort of encouraged to care a lot more about flavor text than humans are. If we can get an admin ruling that this stuff (including the 0th law variations) doesn't count as laws, though, I'll be cool with it.

Law 0 is already treated as a suggestion; malf AIs are allowed to do whatever and don't actually have to work towards completing their objectives.

**Comment by Mothblocks** [comment]
Linters failing, you're missing a newline

**Comment by Coastfront** [comment]
> I'm not 100% a fan of this, as silicons are sort of encouraged to care a lot more about flavor text than humans are. If we can get an admin ruling that this stuff (including the 0th law variations) doesn't count as laws, though, I'll be cool with it.

As with all antag flavors on the tgui im pretty sure theres gonna be a button somewhere that just says "this is a suggestion you dont have to follow this if you dont want to"

**Comment by tralezab** [comment]
okay let me finish this for real

**Review by Kylerace — APPROVED** [review]
dm code is good also based feature

**Pull request merged**